src/EditorFeatures/Core/Shared/Extensions/ITextBufferEditExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
+{
+    internal static class ITextBufferEditExtensions
+    {
+        private static Exception s_lastException = null;
+
+        /// <summary>
+        /// Logs exceptions thrown during <see cref="ITextBufferEdit.Apply"/> as we look for issues.
+        /// </summary>
+        /// <param name="edit"></param>
+        /// <returns></returns>
+        public static ITextSnapshot ApplyAndLogExceptions(this ITextBufferEdit edit)
+        {
+            try
+            {
+                return edit.Apply();
+            }
+            catch (Exception e) when (ErrorReporting.FatalError.ReportWithoutCrash(e))
```

Comment:
Right, the "dialog handler" catches it, so with this change we would be logging the exception in our own telemetry, but we should be able to get this same data already.  If not, I think it would be good to log these exceptions on the editor side